src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -640,6 +637,16 @@ private function cartVoucherHasAmountReduction($cartVoucher)
         return isset($cartVoucher['reduction_amount']) && $cartVoucher['reduction_amount'] > 0;
     }
 
+    /**
+     * @param array $cartVoucher
+     *
+     * @return bool
+     */
+    private function cartVoucherHasGiftProductReduction($cartVoucher)
+    {
+        return array_key_exists('gift_product', $cartVoucher) && $cartVoucher['gift_product'];
```

Comment:
```suggestion
        return !empty($cartVoucher['gift_product']);
```

Is it really needed to add a method only for this check?

Reply:
Not really needed, but just to follow suit with current code.

Reply:
I see what you mean now. Well, the only reason is to make the `if` statement more self-explanatory  ( the naming `cartVoucherHasGiftProductReduction` inside `if` emphasizes the fact that we need the 3 conditions that have "similar" wight). But you know, I am still learning the codebase so I can't tell which one is more suitable until you directly tell me ;-)

Reply:
In fact, this is the same thing.
```
!empty($cartVoucher['gift_product']); // => Not false, 0, '', null, []
array_key_exists('gift_product', $cartVoucher) && $cartVoucher['gift_product'] // => Check if the key exists in the table and is not null, false, 0, '', etc
```

This is the same check, but it's faster due to only one call

Reply:
Thank you very much :). 
If i understand well (because I really have a slow brain these days which I hope for none ), we can we go further by adding [one another commit](https://github.com/PrestaShop/PrestaShop/pull/17502/commits/099a863b16a8a3405af20d28269b483ccb015cb0).